### PR TITLE
python3Packages.PyGithub: 1.54.1 -> 1.55

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 1.55

Change log: https://github.com/PyGithub/PyGithub/releases/tag/v1.55

For #120804 to get `criticality-score` to build.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120847` at accee089 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/08fb84b091e291c01b666753c6012798">cligh</a></li>
    <li><a href="https://gist.github.com/8960cd2026d22ca808fc1152a7db4f04">mathlibtools</a> (python38Packages.mathlibtools)</li>
    <li><a href="https://gist.github.com/dd65b3ec938b8e0d96da8a57ee469904">python38Packages.PyGithub</a></li>
    <li><a href="https://gist.github.com/13ea0bfe1aafb65a4a8bd531373320d8">python38Packages.criticality-score</a></li>
    <li><a href="https://gist.github.com/0dd74c2954f7f15b6c36527f040747bf">python38Packages.howdoi</a></li>
    <li><a href="https://gist.github.com/9bbedd0d8021906811f6b64a076e901b">python38Packages.keep</a></li>
    <li><a href="https://gist.github.com/c448896203e15edddc74223067d57701">python39Packages.PyGithub</a></li>
    <li><a href="https://gist.github.com/c87f3d84ba891ae865c4a90f6dca0288">python39Packages.criticality-score</a></li>
    <li><a href="https://gist.github.com/82bae8a54e8a26e1e6097a44e9aa3e72">python39Packages.howdoi</a></li>
    <li><a href="https://gist.github.com/6f4c16aff5a96a319a4d5d6ec9220fe2">python39Packages.keep</a></li>
    <li><a href="https://gist.github.com/27dd49bb605dced65158215eb6f279aa">python39Packages.mathlibtools</a></li>
    <li><a href="https://gist.github.com/d7316626aa437bacef20643699175bf2">sourcehut.dispatchsrht</a></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/pyGithub/default.nix:35:5:
```
   |
35 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/pyGithub/default.nix:33:0:
```
   |
33 |     description = "A Python (2 and 3) library to access the GitHub API v3";
   | ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by fabaff** [comment]
Requires for some packages #120804 to be merged.

File: pkgs/development/python-modules/pyGithub/default.nix [hunk 1]
```diff
@@ -3,36 +3,41 @@
...
   doCheck = false;
+  pythonImportsCheck = [ "github" ];
 
   meta = with lib; {
+    description = "Python library to access the GitHub API v3";
     homepage = "https://github.com/PyGithub/PyGithub";
-    description = "A Python (2 and 3) library to access the GitHub API v3";
     platforms = platforms.all;
-    license = licenses.gpl3;
+    license = licenses.lgpl3Plus;
```
Comment (mweinelt) [inline pkgs/development/python-modules/pyGithub/default.nix:40]: Can you tell how you found this out?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/pyGithub/default.nix:40]: For example https://github.com/PyGithub/PyGithub/blob/master/github/AccessToken.py#L8

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 120847` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>cligh</li>
    <li>mathlibtools (python38Packages.mathlibtools)</li>
    <li>python38Packages.PyGithub</li>
    <li>python38Packages.criticality-score</li>
    <li>python38Packages.howdoi</li>
    <li>python38Packages.keep</li>
    <li>python39Packages.PyGithub</li>
    <li>python39Packages.criticality-score</li>
    <li>python39Packages.howdoi</li>
    <li>python39Packages.keep</li>
    <li>python39Packages.mathlibtools</li>
    <li>sourcehut.dispatchsrht</li>
  </ul>
</details>

**Pull request merged**